lib/libzpool/uzfs_rebuilding.c

```diff
@@ -366,17 +413,36 @@ uzfs_zvol_create_snaprebuild_clone(zvol_state_t *zv,
 	    strchr(zv->zv_name, '/') + 1,
 	    REBUILD_SNAPSHOT_CLONENAME);
 
+	clone_subname = kmem_asprintf("%s_%s", strchr(zv->zv_name, '/') + 1,
+	    REBUILD_SNAPSHOT_CLONENAME);
+
 	ret = dmu_objset_clone(clonename, snapname);
-	if (ret == EEXIST) {
-		LOG_INFO("Volume:%s already has clone for snap rebuild",
-		    zv->zv_name);
+	if ((ret == EEXIST) || (ret == 0)) {
+		if (ret == EEXIST)
+			LOG_INFO("Volume:%s already has clone for snap rebuild",
+			    zv->zv_name);
+		ret = uzfs_open_dataset(zv->zv_spa, clone_subname, clone_zv);
+		if (ret == 0) {
+			ret = uzfs_hold_dataset(*clone_zv);
+			if (ret != 0) {
+				LOG_ERR("Failed to hold clone: %d", ret);
+				uzfs_close_dataset(*clone_zv);
+				*clone_zv = NULL;
+				uzfs_close_dataset(*snap_zv);
+				destroy_snapshot_zv(zv,
```

Comment:
not sure whether this succeeds as clone still exists

Reply:
Sorry....I could not get this.

Reply:
I mean - as clone is not destroyed, destroy_snapshot may fail